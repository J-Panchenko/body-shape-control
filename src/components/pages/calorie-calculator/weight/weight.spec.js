import { render, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import React from 'react';
import Weight from './weight';

const setup = () => {
  const utils = render(<Weight />);
  const input = utils.getByLabelText('weight-input');
  return {
    input,
    ...utils,
  };
};

describe('Weight component: positive cases', () => {
  const { input } = setup();

  it('It should allow numbers to be inputted', () => {
    fireEvent.change(input, { target: { value: '70' } });
    expect(input.value).toBe('70');
  });

  it('It should not allow letters to be inputted', () => {
    fireEvent.change(input, { target: { value: 'My weight' } });
    expect(input.value).toBe('');
  });

  it('It should allow numbers from 20 to 500 to be inputted', () => {
    fireEvent.change(input, { target: { value: '19' } });
    expect(input.value).toBe('19');
    expect(input.validity.stepMismatch).toBe(false);

    fireEvent.change(input, { target: { value: '501' } });
    expect(input.value).toBe('501');
    expect(input.validity.stepMismatch).toBe(false);
  });

  it('It should be renders correctly', () => {
    const renderer = create(<Weight />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
