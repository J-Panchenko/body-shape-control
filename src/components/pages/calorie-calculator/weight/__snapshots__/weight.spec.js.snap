// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weight component: positive cases matches snapshot 1`] = `
<label
  className="parameter__item"
  htmlFor="weight"
>
  <h2
    className="parameter-subtitle"
  >
    Weight
  </h2>
  <span>
    kg
  </span>
  <input
    aria-label="weight-input"
    className="input"
    id="weight"
    max={500}
    min={20}
    placeholder="0"
    required={true}
    step={0.1}
    type="number"
  />
</label>
`;
